### DISCUSSION TITLE
How to run a global middleware only after a hook is called?

### DISCUSSION THREAD

**Discussion by axelthat** [nuxt/nuxt#17048] [Questions]
I have an `app:created` hook,

```
export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.hook("app:created", async () => {
    const user = useUserStore()
    await user.getUser()
  })
})
```

And a global middleware,
```
export default defineNuxtPlugin(() => {
  addRouteMiddleware(
    "user",
    (to) => {
      const user = useUserStore().user
      if(user.role === 'admin') {
        return true
      }

      return navigateTo('/')
    },
    { global: true }
  )
})
```

However, the middleware runs before the `app:created` hook completes its task. How to make it run only after everything inside `app:created` has been run?

**Comment by procrates** [comment]
Maybe wrap the middleware in app:created like you did in the first plugin?

**Reply by axelthat** [reply]
But I need to call and return `navigateTo` inside the middleware. If I wrap the logic in `app:created`, then I can't call `navigateTo` outside the hook.

**Reply by danielroe** [reply]
You can programmatically add the middleware in the app:created hook.

**Comment by silencerspirit** [comment]
@axelthat try to replace fetch user in app.vue.
